Lesson16/bilge/script.js

```diff
@@ -0,0 +1,416 @@
+  if (temp < 0) {
+    return 'It\'s below freezing.'
+  }
+  else if (temp === 0) {
+    return 'It\'s the freezing point.'
+  }
+  else {
+    return 'It\'s above freezing.'
+  }
+}
+
+console.log('Ex. 7:');
+
+console.log(isItAboveOrBelowFreezing(-1));
+console.log(isItAboveOrBelowFreezing(25));
+console.log(isItAboveOrBelowFreezing(0));
+
+// 8. Write a function that takes in a person's height and weight and checks if they are considered underweight, normal, overweight, or obese using if else statement.
+// Body Mass Index (BMI) is a person’s weight in kilograms (or pounds) divided by the square of height in meters (or feet). A high BMI can indicate high body fatness.
+//Formula: weight (kg) / [height (m)]2
+// Below 18.5	Underweight
+// 18.5 – 24.9	Healthy Weight
+// 25.0 – 29.9	Overweight
+// 30.0 and Above	Obesity
+
+function findBodyMassIndex (weight, height) {
+  let bmi = weight / (height ** 2); 
+
+  if (bmi < 18.5) {
+    return 'The person is underweight.'
+  }
+  else if (bmi >= 18.5 && bmi <= 24.9) {
+    return 'The person is normal.'
+  }
+  else if (bmi >=25 && bmi <= 29.9) {
+    return 'The person is overweight.'
+  }
+  else {
+    return 'The person is obese.'
+  }
+}
+
+console.log('Ex. 8:');
+
+console.log(findBodyMassIndex(40, 1.70));
+console.log(findBodyMassIndex(60, 1.70));
+console.log(findBodyMassIndex(80, 1.70));
+console.log(findBodyMassIndex(200, 1.70));
+
+// 9. Write a function that takes in a person's age and checks if they are a child (age 0-12), teenager (age 13-19), adult (age 20-64), or senior citizen (age 65 and above) using if else statement.
+
+function checkAge(ageOfThePerson) {
+  if (ageOfThePerson >= 0 && ageOfThePerson <= 12) {
+    return 'You are a child.'
+  }
+  else if (ageOfThePerson >= 13 && ageOfThePerson <= 19) {
+    return 'You are a teenager.'
+  }
+  else if (ageOfThePerson >= 20 && ageOfThePerson <= 64) {
+    return 'You are an adult.'
+  }
+  else {
+    return 'You are a senior citizen.'
+  }
+}
+
+console.log('Ex. 9:');
+
+console.log(checkAge(12));
+console.log(checkAge(19));
+console.log(checkAge(64));
+console.log(checkAge(65));
+
+
+// 10. Write a function that takes age as a number and whether they are a student (yes or no). If the age is less than 18 and the user is a student, print "You are a student and not old enough to vote". If the age is between 18 and 65 (inclusive) and the user is not a student, print "You are eligible to vote". If the age is greater than 65, print "You are a senior citizen". If the user is a student but their age is greater than or equal to 18, print "You are old enough to vote".
+
+const student = true;
+
+function checkIfStudentAndVoteEligibility (age, student) {
+  if (age < 18 && student === true) {
+    return 'You are a student and not old enough to vote.'
+  }
+  else if (age >= 18 && student === true) {
+    return 'You are old enough to vote.'
+  }
+  else if (age >= 18 && age <=65 && student === false) {
```

Comment:
You can negate the boolean value if you need to make sure that condition works if the value is false.
```suggestion
  else if (age >= 18 && age <=65 && !student) {
```